mindsdb/integrations/utilities/rag/splitters/file_splitter.py

```diff
@@ -2,80 +2,109 @@
 from typing import Callable, List, Union
 
 from langchain_core.documents import Document
-from langchain_text_splitters import MarkdownHeaderTextSplitter, HTMLHeaderTextSplitter, RecursiveCharacterTextSplitter
+from langchain_text_splitters import (
+    MarkdownHeaderTextSplitter,
+    HTMLHeaderTextSplitter,
+    RecursiveCharacterTextSplitter,
+)
+
+from mindsdb.interfaces.knowledge_base.preprocessing.models import TextChunkingConfig
 
 from mindsdb.utilities import log
 
 DEFAULT_CHUNK_SIZE = 1000
 DEFAULT_CHUNK_OVERLAP = 50
 DEFAULT_MARKDOWN_HEADERS_TO_SPLIT_ON = [
-    ('#', 'Header 1'),
-    ('##', 'Header 2'),
-    ('###', 'Header 3')
+    ("#", "Header 1"),
+    ("##", "Header 2"),
+    ("###", "Header 3"),
 ]
 DEFAULT_HTML_HEADERS_TO_SPLIT_ON = [
-    ('h1', 'Header 1'),
-    ('h2', 'Header 2'),
-    ('h3', 'Header 3'),
-    ('h4', 'Header 4')
+    ("h1", "Header 1"),
+    ("h2", "Header 2"),
+    ("h3", "Header 3"),
+    ("h4", "Header 4"),
 ]
 logger = log.getLogger(__name__)
 
 
 @dataclass
 class FileSplitterConfig:
-    '''Represents configuration needed to split a file into chunks for retrieval.'''
+    """Represents configuration needed to split a file into chunks for retrieval."""
+
     # Target chunk size in characters. Not all splitters will adhere exactly to this (it's more of a guideline)
     chunk_size: int = DEFAULT_CHUNK_SIZE
     # How many characters each chunk should overlap. Not all splitters will adhere exactly to this (it's more of a guideline)
     chunk_overlap: int = DEFAULT_CHUNK_OVERLAP
+    # Chunking parameters are passed as a TextChunkingConfig
+    text_chunking_config: TextChunkingConfig = TextChunkingConfig(
+        chunk_size=DEFAULT_CHUNK_SIZE, chunk_overlap=DEFAULT_CHUNK_OVERLAP
+    )
     # Default recursive splitter to use for text files, or unsupported files
-    recursive_splitter: RecursiveCharacterTextSplitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+    recursive_splitter: RecursiveCharacterTextSplitter = RecursiveCharacterTextSplitter(
```

Comment:
potentially, text_chunking_config is not full initialised, may be better to move to a  `__post_init__`

 ```
 .
 .
recursive_splitter: RecursiveCharacterTextSplitter = None

    def __post_init__(self):
        if self.recursive_splitter is None:
            self.recursive_splitter = RecursiveCharacterTextSplitter(
                chunk_size=self.text_chunking_config.chunk_size,
                chunk_overlap=self.text_chunking_config.chunk_overlap,
                length_function=self.text_chunking_config.length_function,
                separators=self.text_chunking_config.separators,
            )
```